coil-core/src/commonMain/kotlin/coil3/Uri.common.kt

```diff
@@ -29,7 +29,22 @@ class Uri internal constructor(
 val Uri.pathSegments: List<String>
     get() {
         val path = path ?: return emptyList()
-        return path.substring(1).split('/')
+
+        val segments = mutableListOf<String>()
+        var index = 0
+        while (index < path.length) {
+            val startIndex = index + 1
+            index = path.indexOf('/', startIndex = startIndex)
+            if (index == -1) {
+                index = path.length
+            }
+
+            val segment = path.substring(startIndex, index)
+            if (segment.isNotEmpty()) {
+                segments += segment
+            }
+        }
+        return segments
```

Comment:
why not
```suggestion
        return  path.substring(1)
            .split('/')
            .filterNot { it.isEmpty() }
```
?

Reply:
@connyduck Using `split('/')` and `filterNot { it.isEmpty() }` iterates over the data twice and creates two lists. The other solution does one iteration and creates one list.